File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    for i in goal: # 앞자리만 계속 빼주기 
        if cards1 and i == cards1[0]: # cards1이 비어있지 않고 첫 째가 일치하면
            del cards1[0] # del 이 인덱스로 삭제, remove는 값으로 삭제 
        
        elif cards2 and i == cards2[0]:
            del cards2[0]

        else:
            return 'No'
    return 'Yes'

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        List<String> cards1_list = new ArrayList<>(Arrays.asList(cards1));
        List<String> cards2_list = new ArrayList<>(Arrays.asList(cards2));
        
        for(String g : goal){
            if((!cards1_list.isEmpty()) && (g.equals(cards1_list.get(0)))) {
            	cards1_list.remove(0);
            }
            else if((!cards2_list.isEmpty()) && (g.equals(cards2_list.get(0)))) {
            	cards2_list.remove(0);
            }else {
            	return "No";
            }
        }
        
        return "Yes";
    }
}